ntbs-service/Models/SearchParameters.cs

```diff
@@ -14,5 +14,16 @@ public class SearchParameters
         public string IdFilter { get; set; }
         public int? SexId { get; set; }
         public PartialDate PartialDob { get; set; }
+
+        [MinLength(2)]
+        [RegularExpression(ValidationRegexes.CharacterValidation, ErrorMessage = ValidationMessages.StandardStringFormat)]
+        public string FamilyName { get; set; }
+
+        [MinLength(2)]
+        [RegularExpression(ValidationRegexes.CharacterValidation, ErrorMessage = ValidationMessages.StandardStringFormat)]
+        public string GivenName { get; set; }
+
+        public bool SearchParamsExist => !string.IsNullOrEmpty(GivenName) || !string.IsNullOrEmpty(FamilyName) || !string.IsNullOrEmpty(IdFilter) || 
+                                          SexId != null || (PartialDob == null ? true : !PartialDob.IsEmpty());

```

Comment:
`!PartialDob || PartialDob.IsEmpty()`??

Reply:
This will throw an exception if partialDob is null - I'm sure there is a way round this if what I have here and in NotificationSearchBuilder needs to be refactored :)

Reply:
hmmm.. i'd thought if `PartialDob` was `null`, then
`!PartialDob` would be true and the `||` would short-circuit to true, never running `.IsEmpty`? Is that not happening?
Either way, not a massive problem here, just thought it looked a little messy so if we could improve it some that'd be great :)

Reply:
Yep I thought that should happen but it was running it anyway and throwing an exception

Reply:
As helpfully pointed out by Conor, c# doesn't make objects into logic values directly :'(
But I think the points still stands with something like `PartialDob == null || PartialDob.IsEmpty()` as more readable then the ternary operator
(same goes for the other comment!)